src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs

```diff
@@ -72,6 +83,19 @@ public MessageMetadata GetMessageMetadata(string messageTypeIdentifier)
             return null;
         }
 
+        Type GetType(string messageTypeIdentifier)
```

Comment:
I suppose it could have been used, but I didn't think about it. From a perf standpoint, both approaches appear to be the same.